tests/std/tests/P0896R4_views_single/test.cpp

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <memory>
+#include <new>
+#include <ranges>
+#include <string>
+#include <utility>
+
+using namespace std;
+
+template <class>
+constexpr bool always_false = false;
+
+struct only_copy_constructible {
+    int val;
+
+    constexpr explicit only_copy_constructible(int i) noexcept : val{i} {}
+    only_copy_constructible(const only_copy_constructible&) = default;
+    only_copy_constructible(only_copy_constructible&& that) : val{exchange(that.val, -1)} {}
+
+    bool operator==(const only_copy_constructible&) const = default;
+
+    template <class T = void>
+    auto operator&() const {
+        static_assert(always_false<T>);
+    }
+
+    template <class U>
+    auto operator,(U&&) const {
+        static_assert(always_false<U>);
+    }
+};
+
+struct literal_class {
+    int value;
+
+    literal_class() = default;
+    constexpr explicit literal_class(int i) noexcept : value{i} {}
+
+    bool operator==(const literal_class&) const = default;
+};
+
+template <class T, class... Args>
+constexpr bool test_one_type(T value, Args&&... args) {
+    // validate type properties
+    using R = ranges::single_view<T>;
+    static_assert(ranges::view<R> && ranges::contiguous_range<R> && ranges::sized_range<R> && ranges::common_range<R>);
+
+    // validate CTAD and const T& constructor
+    same_as<R> auto r0 = ranges::single_view{value};

```

Comment:
Should include `<concepts>` for `same_as`.

Reply:
Ditto.